### PR TITLE
Introduce Eclipse3DProperties

### PR DESCRIPTION
**Pull request by pgdr**
Introduce Eclipse3DProperties and moved GridProperties stuff to new class.

Clean up initialization process of EclipseState.  Moved some accessors from EclipseState to Eclipse3DProperties that now need to be accessed via EclipseState.getEclipseProperties.

Made some members of state/properties references instead of shared_ptr.

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp [hunk 1]
```diff
@@ -0,0 +1,920 @@
+/*
+  Copyright 2013 Statoil ASA.
```
Comment (atgeirr) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: Year does not seem correct...

File: opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp [hunk 2]
```diff
@@ -0,0 +1,920 @@
...
+#include <opm/parser/eclipse/EclipseState/Tables/TableManager.hpp>
+#include <opm/parser/eclipse/Units/Dimension.hpp>
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+
+namespace Opm {
+
+    namespace GridPropertyPostProcessor {
+
+        void distTopLayer( std::vector<double>&    values,
+                           const TableManager*     tableManager,
```
Comment (atgeirr) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: It looks to me like several functions take pointers, but never check if they are nullptr. If a non-null argument is required it is better to pass it as a const reference. The EclipseGrid argument below is just one example.

File: opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp [hunk 3]
```diff
@@ -0,0 +1,920 @@
...
+        :
+
+          m_defaultRegion("FLUXNUM"),
+          m_deckUnitSystem(deck.getActiveUnitSystem()),
+		  // Note that the variants of grid keywords for radial grids are not
+		  // supported. (and hopefully never will be)
+		  // register the grid properties
+		  m_intGridProperties(eclipseGrid, makeSupportedIntKeywords()),
+		  m_doubleGridProperties(eclipseGrid,
+                        makeSupportedDoubleKeywords(tableManager.get(), &eclipseGrid, &m_intGridProperties))
```
Comment (jokva) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: Minor, but fix indentation please.

File: opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp [hunk 4]
```diff
@@ -0,0 +1,920 @@
...
+        return m_doubleGridProperties;
+    }
+
+    /*
+      1. The public methods getIntGridProperty & getDoubleGridProperty will
+         invoke and run the property post processor (if any is registered); the
+         post processor will only run one time.
+
+         It is important that post processor is not run prematurely, internal
+         functions in EclipseState should therefore ask for properties by
```
Comment (jokva) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: EclipseState -> Eclipse3DProperties

**Comment by akva2** [comment]
Can one of the admins verify this patch?

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Can you please update the pull request description with links to downstream PR's in `opm-material `opm-core` and `opm-autodiff` when they are ready.

**Comment by joakim-hove** [comment]
Please rebase to tip of master.

File: opm/parser/eclipse/EclipseState/Grid/GridPropertyInitializers.cpp [hunk 5]
```diff
@@ -21,39 +22,56 @@ namespace Opm {
...
-            const Deck& d,
-            const EclipseState& state ) :
-        f( fn ), deck( &d ), es( &state )
+            const TableManager* tables,
+            const EclipseGrid* grid,
+            GridProperties<int>* ig_props,
+            GridProperties<double>* dg_props)
+        :
+            f( fn ),
+            // es( &state )
```
Comment (jokva) [inline opm/parser/eclipse/EclipseState/Grid/GridPropertyInitializers.cpp]: Please don't commit out-commented code.

**Comment by pgdr** [comment]
Closing this in favour of  Introduce Eclipse3DProperties #751

**Pull request closed without merging**